.github/workflows/ci_tests.yaml

```diff
@@ -73,7 +73,7 @@ jobs:
             optional-packages: ''
           - python-version: '3.12'
             numpy-version: '1.26'
-            pandas-version: ''
+            pandas-version: '=2.1.4'
```

Comment:
For mysterious reasons, this PR works well for pandas v1.5 (our minimum support version) and v2.1 (see the successful job run at https://github.com/GenericMappingTools/pygmt/actions/runs/8308744195/job/22739307032?pr=2398) but crashes for pandas v2.2 (see the failures at https://github.com/GenericMappingTools/pygmt/actions/runs/8308715874/job/22739243662?pr=2398).

In https://github.com/GenericMappingTools/pygmt/actions/runs/8308715874/job/22739243662?pr=2398, the test `test_grdlandmask_outgrid` crashes, but tests of other module wrappers (e.g., `grdcut`/`grdfill`) still pass.

Reply:
Still don't know what happens, but it seems `grid.sortby()` causes the crash. Found a solution at https://github.com/pydata/xarray/discussions/6695 and apply it in 229bc1e7b6875c95babeae0f1b77c87db0f780d2. Now no crashes.